trl/trainer/dpo_trainer.py

```diff
@@ -815,27 +607,28 @@ def make_inputs_require_grad(module, input, output):
 
             # tokenize the dataset, lower writer batch size to avoid OOM (frequent in vision models)
             fn_kwargs = {
-                "tokenizer": self.processing_class,
-                "args": args,
-                "processor": self.processor if self.is_vision_model else None,
-                "model": model if self.is_encoder_decoder else None,
+                "processing_class": processing_class,
+                "max_prompt_length": args.max_prompt_length,
+                "max_completion_length": args.max_completion_length,
+                # for enc-dec, we add the special tokens ([bos_token] + prompt + [eos_token]; completion + [eos_token])
+                "add_special_tokens": self.is_encoder_decoder,
             }
             train_dataset = train_dataset.map(
-                _tokenize,
+                self.tokenize_row if not self.is_vision_model else self.process_row,
                 fn_kwargs=fn_kwargs,
-                batched=True,
                 num_proc=self.dataset_num_proc,
                 writer_batch_size=10,
                 desc="Tokenizing train dataset",
+                load_from_cache_file=False,
```

Comment:
Is this to prevent the issue we've discussed previously about how changing a data mixture / number of samples doesn't recompute the processing?

Reply:
Arf no, it was more convenient to have it when debugging, good catch
```suggestion
```